### PR TITLE
Add support for Azure Spring Cloud logs

### PR DESCRIPTION
**Pull request by narph**
- Enhancement

**PR description will be updated soon**

## What does this PR do?

Add support for Azure Spring Cloud logs

image

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues

-

Example:

```
 {
        "_index" : ".ds-logs-azure.springcloudlogs-default-2021.08.04-000001",
        "_type" : "_doc",
        "_id" : "B6R6EXsBdXPyPPk9OOPJ",
        "_score" : null,
        "_source" : {
          "agent" : {
            "hostname" : "docker-fleet-agent",
            "name" : "docker-fleet-agent",
            "id" : "dcb938c2-639e-4251-b63e-4e87d8d02cb9",
            "ephemeral_id" : "cf5e0780-ffa3-41b1-afb3-77231370a3b1",
            "type" : "filebeat",
            "version" : "7.14.0"
          },
          "log" : {
            "level" : "INFO"
          },
          "elastic_agent" : {
            "id" : "dcb938c2-639e-4251-b63e-4e87d8d02cb9",
            "version" : "7.14.0",
            "snapshot" : true
          },
          "message" : "Get config server actuator health status UP.",
          "azure-eventhub" : {
            "sequence_number" : 771,
            "consumer_group" : "$Default",
            "offset" : 600952,
            "eventhub" : "insights-logs-systemlogs",
            "enqueued_time" : "2021-08-04T14:04:21.559Z"
          },
          "tags" : [
            "azure-springcloudlogs"
          ],
          "cloud" : {
            "provider" : "azure"
          },
          "input" : {
            "type" : "azure-eventhub"
          },
          "@timestamp" : "2021-08-04T14:03:33.185Z",
          "ecs" : {
            "version" : "1.10.0"
          },
          "data_stream" : {
            "namespace" : "default",
            "type" : "logs",
            "dataset" : "azure.springcloudlogs"
          },
          "host" : {
            "hostname" : "docker-fleet-agent",
            "os" : {
              "kernel" : "4.19.128-microsoft-standard",
              "codename" : "Core",
              "name" : "CentOS Linux",
              "type" : "linux",
              "family" : "redhat",
              "version" : "7 (Core)",
              "platform" : "centos"
            },
            "containerized" : true,
            "ip" : [
              "192.168.0.7"
            ],
            "name" : "docker-fleet-agent",
            "id" : "78315c3233258f2dcd540ed749ab1701",
            "mac" : [
              "02:42:c0:a8:00:07"
            ],
            "architecture" : "x86_64"
          },
          "event" : {
            "agent_id_status" : "verified",
            "ingested" : "2021-08-04T14:03:47.784957800Z",
            "kind" : "event",
            "action" : "Microsoft.AppPlatform/Spring/SystemLogs",
            "dataset" : "azure.springcloudlogs"
          },
          "azure" : {
            "subscription_id" : "....",
            "springcloudlogs" : {
              "operation_name" : "Microsoft.AppPlatform/Spring/SystemLogs",
              "category" : "SystemLogs",
              "event_category" : "Administrative",
              "logtag" : "F",
              "properties" : {
                "stack" : null,
                "service_name" : "obssprincloud",
                "logger" : "com.microsoft.azure.spring.service.controller.DeploymentHealthCheckController",
                "thread" : "XNIO-1 task-1",
                "type" : "ConfigServer"
              }
            },
            "resource" : {
              "provider" : "MICROSOFT.APPPLATFORM/SPRING",
              "name" : "OBSSPRINCLOUD",
              "id" : "/SUBSCRIPTIONS/.../RESOURCEGROUPS/TESTM/PROVIDERS/MICROSOFT.APPPLATFORM/SPRING/OBSSPRINCLOUD",
              "group" : "TESTM"
            }
          }
        },
        "sort" : [
          1628085813185
        ]
      }
```

```
{
        "_index" : ".ds-logs-azure.springcloudlogs-default-2021.08.04-000001",
        "_type" : "_doc",
        "_id" : "XqWZEXsBdXPyPPk9OgOY",
        "_score" : null,
        "_source" : {
          "agent" : {
            "hostname" : "docker-fleet-agent",
            "name" : "docker-fleet-agent",
            "id" : "dcb938c2-639e-4251-b63e-4e87d8d02cb9",
            "type" : "filebeat",
            "ephemeral_id" : "cf5e0780-ffa3-41b1-afb3-77231370a3b1",
            "version" : "7.14.0"
          },
          "log" : {
            "level" : "Informational"
          },
          "elastic_agent" : {
            "id" : "dcb938c2-639e-4251-b63e-4e87d8d02cb9",
            "version" : "7.14.0",
            "snapshot" : true
          },
          "message" : "2021-08-04 14:37:03.902  INFO [thisissecond,,,] 1 --- [trap-executor-0] c.n.d.s.r.aws.ConfigClusterResolver      : Resolving eureka endpoints via configuration",
          "azure-eventhub" : {
            "sequence_number" : 157,
            "consumer_group" : "$Default",
            "offset" : 140536,
            "eventhub" : "insights-logs-applicationconsole",
            "enqueued_time" : "2021-08-04T14:38:13.477Z"
          },
          "tags" : [
            "azure-springcloudlogs"
          ],
          "geo" : {
            "name" : "westeurope"
          },
          "cloud" : {
            "provider" : "azure"
          },
          "input" : {
            "type" : "azure-eventhub"
          },
          "@timestamp" : "2021-08-04T14:37:03.903Z",
          "ecs" : {
            "version" : "1.10.0"
          },
          "data_stream" : {
            "namespace" : "default",
            "type" : "logs",
            "dataset" : "azure.springcloudlogs"
          },
          "host" : {
            "hostname" : "docker-fleet-agent",
            "os" : {
              "kernel" : "4.19.128-microsoft-standard",
              "codename" : "Core",
              "name" : "CentOS Linux",
              "family" : "redhat",
              "type" : "linux",
              "version" : "7 (Core)",
              "platform" : "centos"
            },
            "ip" : [
              "192.168.0.7"
            ],
            "containerized" : true,
            "name" : "docker-fleet-agent",
            "id" : "78315c3233258f2dcd540ed749ab1701",
            "mac" : [
              "02:42:c0:a8:00:07"
            ],
            "architecture" : "x86_64"
          },
          "event" : {
            "agent_id_status" : "verified",
            "ingested" : "2021-08-04T14:37:39.863775400Z",
            "kind" : "event",
            "action" : "Microsoft.AppPlatform/Spring/logs",
            "dataset" : "azure.springcloudlogs"
          },
          "azure" : {
            "subscription_id" : "...",
            "springcloudlogs" : {
              "operation_name" : "Microsoft.AppPlatform/Spring/logs",
              "log_format" : "RAW",
              "category" : "ApplicationConsole",
              "event_category" : "Administrative",
              "logtag" : "F",
              "properties" : {
                "app_name" : "thisissecond",
                "instance_name" : "thisissecond-default-12-696f9485f8-7m6l2",
                "stream" : "stdout",
                "service_name" : "obssprincloud",
                "service_id" : "99070c7524f14eaf970bbdf35f357772"
              }
            },
            "resource" : {
              "provider" : "MICROSOFT.APPPLATFORM/SPRING",
              "name" : "OBSSPRINCLOUD",
              "id" : "/SUBSCRIPTIONS/.../RESOURCEGROUPS/TESTM/PROVIDERS/MICROSOFT.APPPLATFORM/SPRING/OBSSPRINCLOUD",
              "group" : "TESTM"
            }
          }
        },
        "sort" : [
          1628087823903
        ]
      }
```

## Screenshots

image

image

image

### CODE REVIEW

**Comment by narph** [comment]
@sorantis , @hermantmalik , I have added the logs format and dashboards in the description.

**Comment by hemantmalik** [comment]
Hey, @narph those are looking great. Thank you for sharing the update.

File: packages/azure/data_stream/activitylogs/fields/ecs.yml [hunk 1]
```diff
@@ -1,249 +1,28 @@
...
-  multi_fields:
-    - flat_name: user.name.text
-      name: text
-      norms: false
-      type: text
-  name: user.name
-  type: keyword
+- name: ecs.version
+  external: ecs
+  description: ECS version
```
Comment (mtojek) [inline packages/azure/data_stream/activitylogs/fields/ecs.yml:3]: If you load an external field, you don't need to add descriptions here (I'm not sure if we have overriding feature implemented).

File: packages/azure/data_stream/activitylogs/fields/ecs.yml [hunk 2]
```diff
@@ -1,249 +1,28 @@
...
+  description: Service address
+- name: destination
+  external: ecs
+- name: error
+  external: ecs
+- name: message
+  external: ecs
+- name: event
+  external: ecs
+- name: host
```
Comment (mtojek) [inline packages/azure/data_stream/activitylogs/fields/ecs.yml:15]: You have to mention all leaf-fields not just groups. We don't want to import definitions for all fields behind host, but just ones that are actually used. Unfortunately the tool can't figure it out on it's own.

File: packages/azure/data_stream/activitylogs/fields/ecs.yml [hunk 3]
```diff
@@ -1,249 +1,28 @@
-- description: IP address of the client.
-  name: client.ip
```
Comment (mtojek) [inline packages/azure/data_stream/activitylogs/fields/ecs.yml:2]: Fields defining `client` family is here: https://github.com/elastic/ecs/blob/master/generated/beats/fields.ecs.yml#L147

File: packages/azure/data_stream/springcloudlogs/fields/ecs.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+- name: ecs.version
```
Comment (mtojek) [inline packages/azure/data_stream/springcloudlogs/fields/ecs.yml:1]: I guess you don't need this file now.

**Review by mtojek — APPROVED** [review]
Just one comment about the `ecs.yml` file. Otherwise it LGTM, please wait for the CI status.

**Pull request merged**